### ISSUE TITLE
operator: cilium-operator terminates when Kubernetes libraries try to log a non-existent directory

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#7006]
The following error caused the cilium-operator to exit:
```
W0208 21:57:08.151222       1 reflector.go:270] github.com/cilium/cilium/pkg/k8s/client/informers/externalversions/factory.go:115: watch of *v2.CiliumNetworkPolicy ended with: too old resource version: 1273 (3745)
log: exiting because of error: log: cannot create log: open /tmp/cilium-operator.cilium-operator-b49fd8bf7-2mgs8.unknownuser.log.WARNING.20190208-215708.1: no such file or director
```

Related-issues from other consumers of the client-go library have hit this too: https://github.com/containous/traefik/pull/1817

We should implement a similar fix in the cilium-operator.

**Comment by ianvernon** [comment]
Fixed by #7007

**Comment by ianvernon** [comment]
Reopening; fix in #7007 was not enough.

The problem with getting this to work gracefully is that the default golang flags library does not play nicely with the pflags library / cobra. As a stop-gap, we can create the `/tmp/` directory within the `cilium-operator` image to satisfy glog, and come up with a better long-term fix to handle how glog interacts with cilium.

**Comment by ianvernon** [comment]
Update to anyone that his this issue: the issue was that I was trying to configure `glog`, and not `klog`, so the values to configure were not getting configured within the `klog` dependency via `flag.Set`. Explanation here: 
https://github.com/cilium/cilium/pull/7050#issuecomment-463324944

Credit to @aanm for the correct fix.
